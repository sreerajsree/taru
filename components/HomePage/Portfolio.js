import PortfolioItem from './PortfolioItem';

export default function Portfolio({ articles }) {
  const [newestArticle] = articles;
  return (
    <section>
      <div className="flex flex-col items-center w-full mt-16 md:mt-32 mb-0 md:mb-16 font-title">
        <h2 className="text-7xl mb-6 md:mb-10">Portfolio</h2>
        {/* <p className="text-lg my-8">Past. Present. Future.</p> */}
        {[newestArticle].map(({
          title,
          image,
          date,
          description,
          excerpt,
        }, index) => (
          <PortfolioItem
            key={`portfolio-item-${title}`}
            invert={index % 2 === 1}
            title={title}
            image={image}
            date={date}
            description={description}
            excerpt={excerpt}
          />
        ))}
      </div>
    </section>
  );
}
