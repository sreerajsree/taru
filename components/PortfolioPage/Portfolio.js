import { useEffect, useState } from 'react';
import PortfolioItem from './PortfolioItem';
import { calculateNumItemsOnRow } from '../../lib/utils';

export default function Portfolio({ articles }) {
  const [numItemsOnRow, setNumItemsOnRow] = useState(0);

  useEffect(() => {
    const num = calculateNumItemsOnRow(window.innerWidth);
    setNumItemsOnRow(num);
  }, []);
  return (
    <section>
      <div className="flex flex-row flex-wrap justify-center flex-1">
        {articles.map(({
          image,
          title,
          date,
          description,
          excerpt,
          slug,
        }, index) => {
          const onTopRow = index < numItemsOnRow;
          return (
            <PortfolioItem
              key={`portfolio-item-${title}`}
              onTopRow={onTopRow}
              title={title}
              image={image}
              date={date}
              description={description}
              excerpt={excerpt}
              slug={slug}
            />
          );
        })}
      </div>
    </section>
  );
}
