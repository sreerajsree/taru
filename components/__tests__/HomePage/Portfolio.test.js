import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import * as nextRouter from 'next/router';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

import Portfolio from '../../HomePage/Portfolio';

jest.mock('react-use');
nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ route: '/', query: {} }));

describe('Portfolio', () => {
  beforeEach(() => {
    mockAllIsIntersecting(true);
  });
  test('renders and matches snapshot', () => {
    const article = {
      invert: false,
      title: 'title',
      image: 'image.jpg',
      date: '2019-10-10T00:00:00',
      description: 'desc',
      excerpt: 'excerpt',
    };

    const articles = [
      article,
      {
        ...article,
        invert: true,
      },
    ];
    const { asFragment } = render(<Portfolio articles={articles} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
