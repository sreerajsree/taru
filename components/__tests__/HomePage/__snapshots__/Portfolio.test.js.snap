// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio renders and matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="flex flex-col items-center w-full mt-16 md:mt-32 mb-0 md:mb-16 font-title"
    >
      <h2
        class="text-7xl mb-6 md:mb-10"
      >
        Portfolio
      </h2>
      <div
        class="flex mb-8 flex-col-reverse md:flex-row"
      >
        <div
          class="w-full text-center flex"
        >
          <div
            class=" bg-pink-custom px-2 md:px-0 md:py-8 pt-6 pb-10 md:mx-4 md:min-h-1/2 flex items-center justify-center flex-col"
            style="opacity: 0; transform: translateX(-700px) translateZ(0);"
          >
            <h3
              class="text-2xl m-2 md:mx-8 font-title"
            >
              title
            </h3>
            <p
              class="text-base md:text-lg m-2 md:mx-8 md:my-4 font-title"
            >
              <time
                datetime="2019-10-10T00:00:00"
              >
                October, 2019
              </time>
            </p>
            <p
              class="text-base md:text-lg m-2 md:mx-8 md:my-4 font-body"
            >
              desc
            </p>
            <p
              class="text-base md:text-lg m-2 md:mx-8 md:my-4 font-body"
            >
              'excerpt'
            </p>
            <a
              class="px-4 py-2 mt-6 md:mt-8 border-solid border border-black font-body text-base hover:bg-black hover:text-pink-custom ease-in-out transition duration-500"
              href="/portfolio?internalLink=true"
            >
              Go to Portfolio
            </a>
          </div>
        </div>
        <div
          class="w-full md:mx-4"
        >
          <img
            alt="Cover Image for title"
            class="w-full object-cover h-full md:max-h-p-item"
            style="opacity: 0; transform: translateX(700px) translateZ(0);"
          />
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
